### DISCUSSION TITLE
Storybook code coverage in Turbo Monorepo.

### DISCUSSION THREAD

**Discussion by votemike** [vercel/turborepo#7165] [Help] [answered]
[I have created a new Turbo monorepo](https://github.com/votemike/design-system/commits/main/) using [the design-system example](https://github.com/vercel/turbo/tree/main/examples/design-system).  
I then added testing to Storybook following [these instructions](https://storybook.js.org/docs/writing-tests/test-runner). In the root of the repo, it is now possible to run `pnpm run test-storybook` (assuming storybook is running).  
I then [added test coverage](https://github.com/votemike/design-system/commit/a05a5479dd39ee424e9b5580de0d7d87fcc05ada) as per [this](https://storybook.js.org/docs/writing-tests/test-coverage) which I was hoping would show me uncovered lines/files in the packages/ui/src directory by running `pnpm run test-storybook -- --coverage`, but I can't get it to do so. I get a single line called "All files" which only shows zeros.

<img width="822" alt="Screenshot 2024-01-30 at 14 18 15" src="https://github.com/vercel/turbo/assets/3957065/31a3169b-8129-4aff-b60f-f4b483762124">

Is this possible? Is it a limitation of Turbo/Storybook in some way? Is it just not documented how to do such a thing?  
What am I doing wrong?

Thanks in advance.

**Comment by mehulkar** [comment]
Does it work all right if you run storybook _without_ turbo? If so, please file an issue with a reproduction repo!

**Reply by votemike** [reply]
In what way? It works in a normal repo where all code is based off of the root.
If you're asking if it works in a monorepo without Turbo, I don't know. I used Turbo to avoid manually setting up a monorepo.
Or are you talking about using a different set of commands?

**Reply by mehulkar** [reply]
You can use turbo to setup a repo, but `turbo run` is a wrapper around _your_ scripts. Try executing your scripts without `turbo run` to see if the logs are the same. If you are getting the full logs you expect without turbo, this could be a bug in turbo, and we'd really like an issue filed with a reproduction.

**Reply by votemike** [reply]
If I cd in to `apps/docs/`, get storybook running and then run `pnpm run test-storybook -- --coverage`, the same thing happens. I get 0 coverage for "All files".

**Reply by anthonyshew** [reply, marked as answer]
This indicates that there's some misconfiguration in the setup that isn't related to Turborepo. If you're not running a task using `turbo`, then Turborepo is out of the picture!

You'll want to keep debugging or perhaps ask the folks over at Storybook for some help as we aren't the experts there. 😄

**Reply by votemike** [reply]
I have also asked at Storybook to see what they have to say.
I was wondering if it might be that the example turbo repo shouldn't put the .storybook directory and stories in a different directory to the UI that it is storybooking. It seemed odd to me which is why I opened a discussion here too.
